cmd/utils/nodecmd/chaincmd.go

```diff
@@ -74,78 +75,57 @@ func initGenesis(ctx *cli.Context) error {
 	// Make sure we have a valid genesis JSON
 	genesisPath := ctx.Args().First()
 	if len(genesisPath) == 0 {
-		log.Fatalf("Must supply path to genesis JSON file")
+		logger.Crit("Must supply path to genesis JSON file")
 	}
 	file, err := os.Open(genesisPath)
 	if err != nil {
-		log.Fatalf("Failed to read genesis file: %v", err)
+		logger.Crit("Failed to read genesis file", "err", err)
 	}
 	defer file.Close()
 
 	genesis := new(blockchain.Genesis)
 	if err := json.NewDecoder(file).Decode(genesis); err != nil {
-		log.Fatalf("invalid genesis file: %v", err)
+		logger.Crit("Invalid genesis file", "err", err)
 		return err
 	}
-
-	genesis = checkGenesisAndFillDefaultIfNeeded(genesis)
-
-	if genesis.Config.Istanbul != nil {
-		if err := governance.CheckGenesisValues(genesis.Config); err != nil {
-			logger.Crit("Error in genesis json values", "err", err)
-		}
-
-		// Check if governingNode is properly set
-		if strings.ToLower(genesis.Config.Governance.GovernanceMode) == "single" {
-			if istanbulExtra, err := decodeExtra(genesis.ExtraData); err != nil {
-				logger.Crit("Extra data couldn't be decoded. Please check if your extra data is correct", "err", err)
-			} else {
-				var found bool
-				for _, v := range istanbulExtra.Validators {
-					if v == genesis.Config.Governance.GoverningNode {
-						found = true
-						break
-					}
-				}
-				if !found {
-					logger.Crit("GoverningNode is not in the validator list. Please check if your governingNode address is correct")
-				}
-			}
-		}
+	if genesis.Config == nil {
+		logger.Crit("Genesis config is not set")
```

Comment:
Shouldn't it set default configuration accroding to line # 239?

Reply:
The original code allowed creating a new `genesis.Config`. However, it was not a safe way since the new `genesis.Config` requires a `ChainID` as mandatory. So, I invalidate that case.